lib/wings/hydra/works/services/add_file_node_to_file_set.rb

```diff
@@ -45,14 +45,15 @@ def update
 
         # @param [RDF::URI] the identified use of the file (e.g. Valkyrie::Vocab::PCDMUse.OriginalFile, Valkyrie::Vocab::PCDMUse.ThumbnailImage, etc.)
         # @param [true, false] update_existing when true, try to retrieve existing element before building one
-        def find_or_create_file(use, update_existing)
-          current_file = af_file_set.filter_files_by_type(use).first if update_existing
-          unless current_file
-            af_file_set.files.build
-            current_file = af_file_set.files.last
-            Hydra::PCDM::AddTypeToFile.call(current_file, use)
-          end
-          current_file
+        def find_or_create_file(type, update_existing)
+          association = af_file_set.association(type)
+          raise ArgumentError, "you're attempting to add a file to a file_set using '#{type}' association but the file_set does not have an association called '#{type}''" unless association
+          current_file = association.reader if update_existing
+          current_file || association.build
+        end
+
+        def association_type(use)
+          use.first.to_s.split('#').second.underscore.to_sym
```

Comment:
Does the conversion from URI to symbol which allows the original_file association to be created.